### PR TITLE
Add hana_vars and proper sshkey sharing in Trento

### PR DESCRIPTION
**Pull request by mpagot**
Add mandatory hana_vars variables to the Trento config.yaml Add new argument about ssh key to use when calling trento-agent.yaml Ansible playbook

- Verification run: https://openqaworker15.qa.suse.cz/tests/56198

### CODE REVIEW

**Review by lpalovsky — APPROVED** [review]
LGTM

File: lib/qesapdeployment.pm [hunk 1]
```diff
@@ -283,10 +283,11 @@ sub qesap_prepare_env {
     qesap_yaml_replace(openqa_variables => $variables);
     push(@log_files, $paths{qesap_conf_trgt});
 
-    record_info("QESAP conf", "Generating tfvars file");
+    record_info("QESAP conf", "Generating all terraform and Ansible configuration files");
     push(@log_files, "$paths{terraform_dir}/$provider/terraform.tfvars");
     push(@log_files, "$paths{deployment_dir}/ansible/playbooks/vars/hana_media.yaml");
-    push(@log_files, "$paths{deployment_dir}/ansible/playbooks/vars/hana_vars.yaml");
+    my $hana_vars = "$paths{deployment_dir}/ansible/playbooks/vars/hana_vars.yaml";
+    push(@log_files, $hana_vars) if (script_run("test -e $hana_vars") == 0);
```
Comment (alvarocarvajald) [inline lib/qesapdeployment.pm:290]: Shouldn't this check go after `qesap_execute(cmd => 'configure')`?

Comment (mpagot) [reply lib/qesapdeployment.pm:290]: Good catch. Yes, you are right. And the proof is the missing of hana_vars.yaml in the list here: https://openqaworker15.qa.suse.cz/tests/56179#step/cluster_configure/111

**Review by alvarocarvajald — APPROVED** [review]
LGTM

**Pull request merged**